### PR TITLE
[NativeAOT-LLVM] Integrate thread pool into the event loop

### PR DESCRIPTION
**Pull request by SingleAccretion**
Include the Mono sources verbatim and replace an internal call with an equivalent Emscripten call. Do not define `FeaturePortableThreadPool` for single-threaded WASM anymore. Add a test, and fix HelloWasm's shell JS.

Another unimplemented piece bites the dust...

### CODE REVIEW

File: src/tests/nativeaot/Directory.Build.props [hunk 1]
```diff
@@ -12,7 +12,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <DefineConstants Condition="$(TargetArchitecture) == 'wasm'">$(DefineConstants);CODEGEN_WASM</DefineConstants>
+    <DefineConstants Condition="$(TargetsWasm) == 'wasm'">$(DefineConstants);CODEGEN_WASM</DefineConstants>
```
Comment (AustinWise) [inline src/tests/nativeaot/Directory.Build.props]:
```suggestion
    <DefineConstants Condition="'$(TargetsWasm)' == 'true'">$(DefineConstants);CODEGEN_WASM</DefineConstants>
```

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj [hunk 2]
```diff
@@ -331,12 +329,25 @@
...
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\ThreadPoolBoundHandle.Browser.Mono.cs" />
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\PreAllocatedOverlapped.Browser.Mono.cs" />
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\ThreadPool.Browser.Mono.cs" />
+    <!-- TODO-LLVM: the timer queue is a little more dependent on TS, take the PNSE version for now. -->
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\TimerQueue.Wasi.Mono.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsWasi)'=='true'">
-    <Compile Include="System\Threading\ThreadApiCompat.Wasi.cs" />
+  <ItemGroup Condition="'$(TargetsWasi)' == 'true' and '$(FeatureWasmThreads)' != 'true'">
+    <Compile Include="$(MonoBclSourcesRoot)\System\Threading\ThreadPoolBoundHandle.Browser.Mono.cs" />
```
Comment (yowl) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj:345]: This reads a little strange: including "Browser" files in Wasi.  I suppose its unavoidable right now.

**Review by yowl — APPROVED** [review]

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj (hunk 2 above)
Comment (SingleAccretion) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj:345]: The whole implementation is a PNSE stub; I suppose it just wasn't renamed to something like `.WebAssembly.` at the time WASI was being brought up.

**Pull request merged**